source/MaterialXRenderGlsl/TextureBaker.cpp

```diff
@@ -180,62 +199,93 @@ void TextureBaker::writeBakedMaterial(const FilePath& filename, const StringVec&
     DocumentPtr bakedTextureDoc = createDocument();
 
     // Create top-level elements.
-    NodeGraphPtr bakedNodeGraph = bakedTextureDoc->addNodeGraph("NG_baked");
-    GeomInfoPtr bakedGeom = !udimSet.empty() ? bakedTextureDoc->addGeomInfo("GI_baked") : nullptr;
+    const string bakedNodeGraphName = bakedTextureDoc->createValidChildName("NG_baked");
+    NodeGraphPtr bakedNodeGraph = bakedTextureDoc->addNodeGraph(bakedNodeGraphName);
+    const string bakedGeomName = bakedTextureDoc->createValidChildName("GI_baked");
+    GeomInfoPtr bakedGeom = !udimSet.empty() ? bakedTextureDoc->addGeomInfo(bakedGeomName) : nullptr;
     if (bakedGeom)
     {
         bakedGeom->setGeomPropValue("udimset", udimSet, "stringarray");
     }
-    MaterialPtr bakedMaterial = bakedTextureDoc->addMaterial("M_baked");
-    ShaderRefPtr bakedShaderRef = bakedMaterial->addShaderRef(_shaderRef->getName() + "_baked", _shaderRef->getAttribute("node"));
+    NodePtr bakedShader = bakedTextureDoc->addNode(_shader->getCategory(), _shader->getName() + BAKED_POSTFIX, _shader->getType());
     bakedNodeGraph->setColorSpace(_colorSpace);
 
-    // Create bind elements on the baked shader reference.
-    for (ValueElementPtr valueElem : _shaderRef->getChildrenOfType<ValueElement>())
+    // Add a material node if any specified and connect it to the new shader node
+    if (_material)
```

Comment:
Optional material node creation if the shader node to sample came with a material node specified.